zookeeper-server/src/main/java/org/apache/zookeeper/server/auth/X509AuthenticationProvider.java

```diff
@@ -259,9 +263,15 @@ private String matchAndExtractSAN(X509Certificate clientCert)
         Pattern extractPattern = Pattern.compile(extractRegex);
         Matcher matcher = extractPattern.matcher(matched.iterator().next().get(1).toString());
         if (matcher.find()) {
-            String result = matcher.group();
+            String result;
+            if (extractMatcherGroupIndex != null) {
+                result = matcher.group(extractMatcherGroupIndex);
```

Comment:
This can lead to an unhandled `IndexOutOfBoundsException` if there is no capturing group in the pattern with the given index.

Reply:
I've broadened the try-catch outside of this function to catch all Exceptions. Good eye!